test/likelihoods/test_noise_models.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python3
+
+import unittest
+
+import torch
+
+import gpytorch
+from gpytorch.likelihoods import HeteroskedasticNoise
+
+
+class NumericallyUnstableModelExample(gpytorch.models.GP):
+    def __init__(self):
+        super(NumericallyUnstableModelExample, self).__init__()
+        self.fail_arithmetic = False
+
+    def train(self, mode=True):
+        if mode:
+            self.fail_arithmetic = False  # reset on .train()
+        super().train(mode=mode)
+
+    def forward(self, x):
+        if self.fail_arithmetic:
+            raise ArithmeticError()
+        return gpytorch.distributions.MultivariateNormal(torch.tensor([-3.0]), torch.tensor([[2.0]]))
+
+
+class TestNoiseModels(unittest.TestCase):
+    def test_heteroskedasticnoise_error(self):
```

Comment:
Looks like this is just testing the `NumericallyUnstableModelExample` but not the actual code of the noise model?

Reply:
The issue was that the previous code didn't actually test that things were reset back, seems to be fixed now.